Content Service Example first commit

judtoro opened the pull request:
**Description:**
Project first commit
**Implementation:**
Project was created
**US:**
https://foxites.atlassian.net/browse/PLAT-507
**Test:**
Run Junit Tests
**Documentation:**
N/A

judtoro committed:
Content Service Example first commit

hectormaoglb commented on pom.xml:
```diff
@@ -0,0 +1,188 @@
+
+    <groupId>com.fox.platform</groupId>
+    <artifactId>afa</artifactId>
+    <version>1.0</version>
```
Si es un feature que va a la rama develop debemos manejar la version con el sufijo -SNAPSHOT para este caso 1.0-SNAPSHOT

AlejandraGlob replied:
Se cambia por 1.0-SNAPSHOT

hectormaoglb commented on pom.xml:
```diff
@@ -0,0 +1,188 @@
+    <packaging>jar</packaging>
+
+    <name>afa</name>
+    <url>https://github.com/FOXITES/BaselineVertx</url>
```
Esta no es la URL del proyecto ....

AlejandraGlob replied:
Se cambia por https://github.com/judtoro/contentServiceExample

hectormaoglb commented on pom.xml:
```diff
@@ -0,0 +1,188 @@
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <vertx.version>3.4.1</vertx.version>
+        <newrelic.dropwizard.version>3.4.1</newrelic.dropwizard.version>
+        <libs.version>3.1-SNAPSHOT</libs.version>
```
Ya esta el release de las libs el 5.0 no debemos usar las versiones SNAPSHOT de las lib

AlejandraGlob replied:
Pendiente de permisos para usar la versión 5.0

AlejandraGlob replied:
Listo !!!

hectormaoglb commented on pom.xml:
```diff
@@ -0,0 +1,188 @@
+
+    <dependencies>
+        <dependency>
+            <groupId>com.jayway.jsonpath</groupId>
```
Creo que esa dependencia quedo obsoleta, no es recomendable el uso de jsonpath

AlejandraGlob replied:
Se elimina dependencia y código asociado

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/dom/ent/Channel.java:
```diff
@@ -0,0 +1,62 @@
+package com.fox.platform.cntsrvex.dom.ent;
+
+public class Channel {
```
- javadoc

- Si  esto es un pojo o entidad que se serializa y des-serializa desde un json, es importante colocarle las anotaciones de @JsonIgnoreProperties(ignoreUnknown = true) para que si llegan atribuos en el json que aun no estan en el objeto javano genere una excepción

- En este tipo de objetos es recomendable usar lombok para evitarse la escritura "cansona" de los getter y setter
https://projectlombok.org/

judtoro replied:
La clase Channel finalmente no se usa en el proyecto. Se eliminó.

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/dom/ent/JsonFields.java:
```diff
@@ -0,0 +1,21 @@
+package com.fox.platform.cntsrvex.dom.ent;
+
+public enum JsonFields {
```
Javadoc

judtoro replied:
Corregido

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/exc/RequestException.java:
```diff
@@ -0,0 +1,18 @@
+package com.fox.platform.cntsrvex.infra.exc;
+
+public class RequestException extends RuntimeException {
```
Javadoc

judtoro replied:
Corregido

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/hndlr/HandlersChannelImpl.java:
```diff
@@ -0,0 +1,145 @@
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+public class HandlersChannelImpl {
```
javadoc

judtoro replied:
Corregido

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/hndlr/HandlersChannelImpl.java:
```diff
@@ -0,0 +1,145 @@
+
+public class HandlersChannelImpl {
+
+  private static Logger logger = LogManager.getLogger();
```
Vertx provee un factory y un Logger propios que basa su funcionamiento en SLF4J http://vertx.io/docs/vertx-core/java/#_logging favor cambiar esta implementación

AlejandraGlob replied:
Se usa el logger de Vertx

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/hndlr/HandlersChannelImpl.java:
```diff
@@ -0,0 +1,145 @@
+            .end(request.encodePrettily());
+      });
+
+    } catch (EncodeException e) {
```
Se realizo algun prueba que hiciera throw de este tipo de excepción, este try{ } catch(){} no deberia ir en el interior del lambda del body handler

AlejandraGlob replied:
``` java
    } catch (RequestException e) {
      routingContext.response().setStatusCode(e.getCode())
          .putHeader(HttpHeaders.CONTENT_TYPE, ContentType.TEXT_PLAIN.getMimeType())
          .end(e.getMessage());
    } catch (Exception e) {
      routingContext.response().setStatusCode(HttpStatus.SC_INTERNAL_SERVER_ERROR)
          .putHeader(HttpHeaders.CONTENT_TYPE, ContentType.TEXT_PLAIN.getMimeType())
          .end(e.getMessage());
    }
```
Se cambia por esas dos que pudimos comprobar son utilizadas.

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/hndlr/HandlersChannelImpl.java:
```diff
@@ -0,0 +1,145 @@
+    try {
+
+      Optional<String> countryOpt = Optional.ofNullable(routingContext.request().getParam("countryId"));
+      String countryId = countryOpt.orElseThrow(() -> new RequestException(HttpStatus.SC_BAD_REQUEST, "Bad countryId code"));
```
Este optional podria ir en una sola instrucción (linea 52 y 53), nosotros lo trabajamos así lo hace mas legible...

judtoro replied:
Se hizo el cambio para manejarse en una sola línea

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/hndlr/HandlersChannelImpl.java:
```diff
@@ -0,0 +1,145 @@
+        if (resp.succeeded()) {
+          logger.info("When print !!! {} ", resp.result().body());
+          routingContext.response().setStatusCode(HttpStatus.SC_OK);
+          routingContext.response().end(resp.result().body().toString());
```
y si el body no es un Json ??? no seria mejor por seguridad tener un try - catch dentro del lambda

AlejandraGlob replied:
Se revisó y el body no llega null como para causar una excepción al momento de obtener el toString().

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/hndlr/HandlersChannelImpl.java:
```diff
@@ -0,0 +1,145 @@
+    try {
+
+      Optional<String> countryOpt = Optional.ofNullable(routingContext.request().getParam("countryId"));
+      String countryId = countryOpt.orElseThrow(() -> new RequestException(HttpStatus.SC_BAD_REQUEST, "Bad countryId code"));
```
Optional de obtener el contryId en una sola instrucción

judtoro replied:
Se hizo el cambio para manejarse en una sola línea

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/hndlr/HandlersChannelImpl.java:
```diff
@@ -0,0 +1,145 @@
+      logger.info("CountryId {}", countryId);
+
+      vertx.eventBus().send("get_channels_juan", countryId, resp -> {
+
```
no seria bueno tener un try-catch aqui adentro ???

AlejandraGlob replied:
Probablemente cuando se tenga una mayor lógica, de momento no es necesario.

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/hndlr/HandlersChannelImpl.java:
```diff
@@ -0,0 +1,145 @@
+      vertx.eventBus().send("get_channels_juan", countryId, resp -> {
+
+        if (resp.succeeded()) {
+          logger.info("When print !!! {} ", resp.result().body());
```
si el rsp.result() es nulo ??? puede ocurrir !!!

AlejandraGlob replied:
Se probó el caso de un código de país inexistente y enviando excepciones dentro de la petición. El resultado no ha llegado a ser nulo.

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticle.java:
```diff
@@ -0,0 +1,42 @@
+
+    Router router = getRouter();
+
+    vertx.createHttpServer().requestHandler(router::accept).listen(Integer.getInteger("http.port", 8080), result -> {
```
- Integer.getInteger ??? esto siempre devuelve 8080, creo que aqui debe ir el tema de la configuración ....
- nos podemos ahorrar este lambda con simplemente enviar startFuture.completer()

judtoro replied:
Se obtiene configuración de archivo default-config.json

judtoro replied:
Se hizo el cambio para obtener la configuración desde el archivo default-config.json

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticle.java:
```diff
@@ -0,0 +1,42 @@
+
+  private Router getRouter() {
+    Router router = Router.router(vertx);
+    HandlersChannelImpl handler = new HandlersChannelImpl(vertx);
```
Este objeto es suceptible de que se inyecte por google guice

judtoro replied:
Se hizo la inyección de dependencias con Guice

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticle.java:
```diff
@@ -0,0 +1,42 @@
+import io.vertx.ext.web.Router;
+
+
+public class EndpointVerticle extends AbstractVerticle {
```
javadoc

judtoro replied:
Corregido

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,97 @@
+import io.vertx.ext.web.client.WebClient;
+
+
+public class ProxyChannelsVerticle extends AbstractVerticle {
```
javadoc

judtoro replied:
Corregido

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,97 @@
+  @Override
+  public void start(Future<Void> startFuture) throws Exception {
+
+    vertx.eventBus().consumer("get_channels_ale", this::onMessage);
```
dirección por config

judtoro replied:
Se trae la dirección desde la configuración

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,97 @@
+      doPetition(jsonQuery, resultObj);
+
+    } catch (Exception e) {
+      logger.error("Error getting channels fields.", e);
```
Esto deberia fallar un futuro ... o por lo menos retornar un mensaje de error (message.fail(ex)) por que el otro verticle esta esperando una respuesta

AlejandraGlob replied:
Corregido.

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,97 @@
+    }
+  }
+
+  private void doPetition(JsonObject jsonObj, final Future<JsonObject> resultObj) {
```
metodos que no son obvios (!= get or set) debemos agregar javadoc

judtoro replied:
Corregido

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,97 @@
+    WebClient webclient = WebClient.create(vertx);
+
+    webclient
+        .post(443, "search-omnix-services-sh2266ar6ket7lqcnhj3dpzccu.us-east-1.es.amazonaws.com",
```
parámetros que deben estar en el config

judtoro replied:
Se hizo el cambio para obtener la configuración desde el archivo default-config.json

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,97 @@
+        });
+  }
+
+  private Future<JsonObject> processResponse(Message<String> message) {
```
javadoc

veo un poco enrredado el manejo de este futuro, no es claro, podriamos discutirlo acerca de cual debe ser el mejor aproach

judtoro replied:
Corregido javadoc

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,97 @@
+    return resultObj;
+  }
+
+  public Future<JsonObject> jsonPathResponse(Message<String> message) {
```
igual que lo anterior

judtoro replied:
Corregido javadoc

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticleJuan.java:
```diff
@@ -0,0 +1,68 @@
+import io.vertx.ext.web.client.WebClient;
+
+
+public class ProxyChannelsVerticleJuan extends AbstractVerticle {
```
javadoc

judtoro replied:
Corregido

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticleJuan.java:
```diff
@@ -0,0 +1,68 @@
+
+public class ProxyChannelsVerticleJuan extends AbstractVerticle {
+
+  private static final Logger logger = LogManager.getLogger();
```
logger de vertx

AlejandraGlob replied:
Corregido

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticleJuan.java:
```diff
@@ -0,0 +1,68 @@
+    try {
+      logger.info("Handling message. CountryId: {}.......", message.body());
+
+      String jsonString =
```
deberia ir en la configuración

judtoro replied:
Se hizo el cambio respectivo para tomar la configuración desde el archivo default-config.json

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticleJuan.java:
```diff
@@ -0,0 +1,68 @@
+      WebClient webclient = WebClient.create(vertx);
+
+      webclient
+          .post(443, "search-omnix-services-sh2266ar6ket7lqcnhj3dpzccu.us-east-1.es.amazonaws.com",
```
deberia ir en la configuracion los parametros del webclient

judtoro replied:
Se hizo el cambio respectivo para tomar la configuración desde el archivo default-config.json

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticleJuan.java:
```diff
@@ -0,0 +1,68 @@
+          });
+
+    } catch (Exception e) {
+      logger.error("Error getting channels fields.", e);
```
retornar un message.fail el otro verticle esta esperando respuesta

AlejandraGlob replied:
Se juntaron los Channels y se maneja la observación en la nueva clase.

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/util/QueryUtil.java:
```diff
@@ -0,0 +1,24 @@
+package com.fox.platform.cntsrvex.infra.util;
+
+/**
+ * @author alejandra.ramirez
```
Completar javadoc

judtoro replied:
Corregido

judtoro replied:
Corregido

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/util/QueryUtil.java:
```diff
@@ -0,0 +1,24 @@
+  }
+
+  public static String getChannels(String countryId) {
+    return "{\"query\":{\"bool\":{\"must\":[{\"term\":{\"type.description\":\"olympicschannel\"}},"
```
esto deberia ir en configuración

judtoro replied:
Se hizo el cambio respectivo para tomar la configuración desde el archivo default-config.json

hectormaoglb commented on src/main/resources/com/fox/platform/cntsrvex/infra/util/query.properties:
```diff
@@ -0,0 +1 @@
+query.channels = {"query":{"bool":{"must":[{"term":{"type.description":"olympicschannel"}},{"nested":{"path":"groups","query":{"bool":{"must":[{"nested":{"path":"groups.feeds","query":{"match":{"groups.feeds.countryId":"{0}"}}}}]}},"inner_hits":{}}}]}},"sort":[{"groups.fields.id.raw":{"nested_path":"groups","order":"asc"}}]}
```
La configuración en vertx se hace por medio de JsonObject, no usamos properties

judtoro replied:
Se reemplaza por configuración en default-config.json

hectormaoglb commented on src/test/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticleTest.java:
```diff
@@ -0,0 +1,65 @@
+    vertx = Vertx.vertx();
+    vertx.deployVerticle(ProxyChannelsVerticleJuan.class.getName());
+    vertx.deployVerticle(ProxyChannelsVerticle.class.getName());
+    vertx.deployVerticle(EndpointVerticle.class.getName());
```
falto completar el test context

AlejandraGlob replied:
Se agrega asyncAssertSuccess al último deploy.

hectormaoglb commented on src/test/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticleTest.java:
```diff
@@ -0,0 +1,65 @@
+    vertx.createHttpClient().getNow(port, "localhost", "/channels_ale?countryId='CO'", response -> {
+      ctx.assertEquals(response.statusCode(), HttpStatus.SC_OK);
+      response.bodyHandler(body -> {
+        ctx.assertEquals(body.toJsonArray().size(), 4);
```
tratar de encerrar estas pruebas en un try-catch y hacer un fail, ya que si ocurre una excepción la prueba no falla y no termina

AlejandraGlob replied:
Comprobando y corregido que se queda esperando.

hectormaoglb commented on src/test/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticleTest.java:
```diff
@@ -0,0 +1,65 @@
+  private int port = Integer.getInteger("http.port", 8080);
+
+   @Before
+  public void setUp(TestContext ctx) {
```
javadoc, que se hace en el setup

judtoro replied:
Corregido

hectormaoglb commented on src/test/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticleTest.java:
```diff
@@ -0,0 +1,65 @@
+  }
+
+   @Test
+  public void testRESTMethodsAle(TestContext ctx) {
```
javadoc que hace este test

judtoro replied:
Corregido

hectormaoglb reviewed (changes requested):
Lo veo bien, vamos por buen camino, favor atender notas

MaguilaGlb commented on src/main/resources/cluster.xml:
```diff
@@ -0,0 +1,157 @@
+      <ports>0</ports>
+    </outbound-ports>
+    <join>
+      <multicast enabled="true">
```
false

judtoro replied:
Corregido

jgarridoglb commented on src/main/java/com/fox/platform/cntsrvex/dom/ent/Channel.java:
```diff
@@ -0,0 +1,62 @@
+package com.fox.platform.cntsrvex.dom.ent;
+
+public class Channel {
```
Validar si el campo enabled lo pueden manejar como un boolean

judtoro replied:
Esta clase no se usaba en el proyecto. Se eliminó.

jgarridoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/exc/RequestException.java:
```diff
@@ -0,0 +1,18 @@
+package com.fox.platform.cntsrvex.infra.exc;
+
+public class RequestException extends RuntimeException {
```
¿Esta custom exception no manejaria el StackTrace cuando ocurra un error?. El constructor solo recibe el message y un code .

AlejandraGlob replied:
Corregido.

jgarridoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/hndlr/HandlersChannelImpl.java:
```diff
@@ -0,0 +1,145 @@
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+public class HandlersChannelImpl {
```
De acuerdo a lo que revisamos en la sync no seria necesario tener un HandlerImpl si solo vamos a reenviar la informacion del APIverticle al ProxyVerticle. Podrian unificar los metodos a manera de ejemplo getChannels es un buen nombre para el metodo.

AlejandraGlob replied:
Se unifica el handler con el verticle.

jgarridoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/hndlr/HandlersChannelImpl.java:
```diff
@@ -0,0 +1,145 @@
+    try {
+
+      Optional<String> countryOpt = Optional.ofNullable(routingContext.request().getParam("countryId"));
+      String countryId = countryOpt.orElseThrow(() -> new RequestException(HttpStatus.SC_BAD_REQUEST, "Bad countryId code"));
```
Para el manejo de error, si el CMS no esta disponible por ejemplo la URL a ElasticSearch no es valida o se genera timeout de conexion al elastic Search. Se esta tratando estos escenarios de infraestructura en el error handler.

judtoro replied:
Se actualizó a como se manejan los optional en el proyecto.

jgarridoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,97 @@
+    return resultObj;
+  }
+
+  public Future<JsonObject> jsonPathResponse(Message<String> message) {
```
Dejar un solo metodo para procesar la respuesta se tiene codigo jsonPathResponse y processResponse. De lo que se habia revisado se iba usar serializacion direcatmente a JsonObject.

AlejandraGlob replied:
Se elimina el método jsonPathResponse y su dependencia en el pom.

jgarridoglb reviewed (commented):
comentarios sobre el PR

jgarridoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticle.java:
```diff
@@ -0,0 +1,42 @@
+
+  private Router getRouter() {
+    Router router = Router.router(vertx);
+    HandlersChannelImpl handler = new HandlersChannelImpl(vertx);
```
Se propone que los valores literales por ejemplo "/channels_ale" sean private static final String GET_CHANNELS = "/channels"

judtoro replied:
**OK**

jgarridoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,97 @@
+    return resultObj;
+  }
+
+  public Future<JsonObject> jsonPathResponse(Message<String> message) {
```
¿Como se maneja el error si el metodo falla en la trsnsformacion de la informacion?

jgarridoglb commented on src/test/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticleTest.java:
```diff
@@ -0,0 +1,65 @@
+  private int port = Integer.getInteger("http.port", 8080);
+
+   @Before
+  public void setUp(TestContext ctx) {
```
La unit test es una prueba de integracion por lo tanto en el package o en el javadoc referenciar que es una integration test.

judtoro replied:
**OK**

jgarridoglb commented on src/test/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticleTest.java:
```diff
@@ -0,0 +1,65 @@
+    vertx.createHttpClient().getNow(port, "localhost", "/channels_ale?countryId='CO'", response -> {
+      ctx.assertEquals(response.statusCode(), HttpStatus.SC_OK);
+      response.bodyHandler(body -> {
+        ctx.assertEquals(body.toJsonArray().size(), 4);
```
Podrias validar de la siguiente forma para que no quede el tamaño del array de la respuesta en duro para la prueba de la siguiente forma:
context.assertNotNull( body.toJsonArray() );
context.assertFalse( body.toJsonArray().isEmpty() );

judtoro replied:
Se hizo el cambio.

jgarridoglb commented on src/test/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticleTest.java:
```diff
@@ -0,0 +1,65 @@
+    vertx.createHttpClient().getNow(port, "localhost", "/channels_ale?countryId='CO'", response -> {
+      ctx.assertEquals(response.statusCode(), HttpStatus.SC_OK);
+      response.bodyHandler(body -> {
+        ctx.assertEquals(body.toJsonArray().size(), 4);
```
Validar este javadoc para cada metodo de la prueba 	
/**
	 * Integration test with the Scenario
	 * SCENARIO: UpdateLevelById
	 * GIVEN a mediaID valid
	 * AND   a authlevel valid 
	 * WHEN execute the method onMessage(Verticle) 
	 * THEN Return a UpdateLevelResponse not NULL
	 * 1. test that the response is NOT NULL
	 * 2. Test that the streamID is NOT NULL.
	 * 3. Test that the authLevelOld is NOT NULL.
	 * @param context
	 */

judtoro replied:
Se usó el formato sugerido

jgarridoglb reviewed (commented):
comentarios sobre el PR

judtoro committed:
PR review: Javadoc

Commit:
Observaciones: pom.xml y logs

Observaciones del pom.xml y los logs.

Commit:
Constantes

Se asigna la constante ContentType.TEXT_PLAIN.getMimeType()
 para el texto text/plain

Commit:
json-path

Se corrigen las observaciones de json-path

judtoro committed:
Se actualiza documentación

Commit:
libs a version 5.0

judtoro committed:
Se usa Guice para injectar dependencia a la clase HandlersChannel

judtoro committed:
Merge branch 'feature/PLAT-507' of https://github.com/judtoro/contentServiceExample into feature/PLAT-507

judtoro committed:
Comentarios sobre uso de Optional

Commit:
Manejo de Excepciones

Se toma más detalle en el manejo de excepciones y se unifican los dos canales, Juan y Ale.

judtoro committed:
Avances configuración

judtoro committed:
Implementación configuración

judtoro committed:
Avances configuración

Commit:
try-catch

Se agrega el try-catch en caso de error durante la prueba.

judtoro committed:
Multicast = false

judtoro committed:
Merge branch 'feature/PLAT-507' of https://github.com/judtoro/contentServiceExample into feature/PLAT-507

judtoro committed:
Actualizacion

judtoro committed:
Comentarios

judtoro committed:
Name conventions

Commit:
Se quita funcionamiento del Handler

Se elimina la clase HandlersChannel para evitar pasar la configuraci[on del verticle.

judtoro committed:
Se agrega path de endpoint a la configuración

Commit:
asyncAssertSuccess

Se corrige observación de la falta de asyncAssertSuccess para el último verticle.

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticle.java:
```diff
@@ -0,0 +1,99 @@
+        if (resp.succeeded()) {
+          logger.info("Result " + resp.result().body());
+          routingContext.response().setStatusCode(HttpStatus.SC_OK);
+          routingContext.response().end(resp.result().body().toString());
```
el body podria ser null ? si eso sucede esto estalla en atomos volando .....

AlejandraGlob replied:
Siempre se regresa un arreglo, aunque este se encuentre vacío.
Dentro de las pruebas se consideró un countryId inexistente y un try - catch en caso de que ocurra algún error en la consulta.

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,133 @@
+      getChannelsFromCMS(jsonQuery, resultObj);
+
+    } catch (Exception e) {
+      logger.error("Error getting channels fields.", e);
```
en este catch no deberiamos responder el message con un fail ????

judtoro replied:
**OK**

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,133 @@
+
+  }
+
+  private void onMessage(Message<String> message) {
```
javadoc

judtoro replied:
**OK**

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,133 @@
+   * @param message: Message object used to reply the response from Omnix.
+   * @return
+   */
+  private Future<JsonObject> processResponse(Message<String> message) {
```
Este manejo de los Future aunque en la parctica funciona, para mi es un poco confuso, por que ...
el proceso de la respuesta del web client que deberia hacerse despues, esta en un proceso anterior, yo lo haria de otra forma, donde llamo el web client retorno un futuro y ese futuro lo mando a un process que esta despues, esto lo hace mas natural a la lectura

hectormaoglb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/ProxyChannelsVerticle.java:
```diff
@@ -0,0 +1,133 @@
+   * @return
+   */
+  public String getPayload(String payload, String countryId) {
+    return payload.replace("${countryId}", countryId);
```
Para este tipo de opraciones usamos StrSubstitutor

judtoro replied:
**OK**

hectormaoglb commented on src/test/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticleTest.java:
```diff
@@ -0,0 +1,198 @@
+
+    vertx = Vertx.vertx();
+    vertx.deployVerticle(ProxyChannelsVerticle.class.getName(), options);
+    vertx.deployVerticle(EndpointVerticle.class.getName(), options, ctx.asyncAssertSuccess());
```
aqui como se lanzan 2 verticles y este es de tipo asincrono quizas (caso hipotetico) se pueda hacer un complete del contexto del setup sin que el verticle ProxyChannelsVerticle se haya desplegado en su totalidad, les recomiendo usar una composicion de futuros para esperar al primer despliegue despues el segundo y por ultimo se completa el contexto

hectormaoglb commented on src/test/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticleTest.java:
```diff
@@ -0,0 +1,198 @@
+        try {
+          expected.accept(body);
+        } catch (Exception e) {
+          log.error("Body {" + body + "}", e);
```
aqui no deberiamos fallar la prueba ...
ctx.fail()
en vez de hacer un logger.error

AlejandraGlob replied:
Corregido.

hectormaoglb reviewed (commented):
Muchos menos comentarios, favor verificarlos

Commit:
Observaciones

Se corrige la observación en el método testRESTMethods para enviar un ctx.fail.

Se cambia cadena "UtF-8" por constante.
Se evita devolver una cadena vacia y se transforma por una excepción de tipo Runtime.

judtoro committed:
Using StrSubstitutor

MaguilaGlb commented on README.md:
```diff
@@ -1 +1,4 @@
-# contentServiceExample
\ No newline at end of file
+# contentServiceExample
```
para un MS real usamos este readme.md como template https://github.com/FOXITES/BaselineVertx/blob/master/README.md

judtoro replied:
**OK**

MaguilaGlb commented on src/main/java/com/fox/platform/cntsrvex/dom/ent/JsonFields.java:
```diff
@@ -0,0 +1,31 @@
+ * @author juan.toro
+ *
+ */
+public enum JsonFields {
```
estan usando formatter de google?
si no lo estan usando sigan estas instrucciones
http://www.practicesofmastery.com/post/eclipse-google-java-style-guide/
la idea es comenzar a usarlo en los ms nuevos
pls agreguen esto a la docu del ramp up

judtoro replied:
- En este momento se está usando y se verificó la configuración contra lo la guía adjunta... como debería quedar?, sin líneas en blanco?
- Ya se añadió a la documentación de Ramp Up

MaguilaGlb replied:
@judtoro fijate la linea 19 y 20 no estan alineadas y si me bajo el codigo y aplico el formatter acomoda los espacios, avisa cualquier cosa

MaguilaGlb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticle.java:
```diff
@@ -0,0 +1,99 @@
+ * 1. Get channels (Alejandra implementation): /channels_ale?countryId={country}
+ * 2. Get channels (Juan implementation): /channels_juan?countryId={country}
+ *
+ * &#64;author juan.toro
```
porque esos caracteres &#64?

judtoro replied:
**OK**

MaguilaGlb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticle.java:
```diff
@@ -0,0 +1,99 @@
+
+      vertx.eventBus().send(contentServiceExampleConfig.getAddress(), countryId, resp -> {
+
+        if (resp.succeeded()) {
```
encapsularia en un metodo el manejo de routingContext.response() para tocar siempre en un único lugar y no estar buscando por el codigo donde se maneja la resp

AlejandraGlob replied:
**Ok**

MaguilaGlb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticle.java:
```diff
@@ -0,0 +1,99 @@
+          routingContext.response().setStatusCode(HttpStatus.SC_OK);
+          routingContext.response().end(resp.result().body().toString());
+        } else {
+          logger.error("Error trying to reach Omnix!", resp.cause());
```
tiraria una excepcion y dejaria que el catch maneje el response

judtoro replied:
**OK**

MaguilaGlb commented on src/main/java/com/fox/platform/cntsrvex/infra/serv/EndpointVerticle.java:
```diff
@@ -0,0 +1,99 @@
+      });
+
+    } catch (RequestException e) {
+      routingContext.response().setStatusCode(e.getCode())
```
en los catch agregaria logger.error para saber por consola que pasó y reusaria un metodo para manejar el response

judtoro replied:
**OK**

judtoro committed:
Se revisan comentarios en el PR

judtoro committed:
Fix

Commit:
PR observaciones

MaguilaGlb:
encapsularia en un metodo el manejo de routingContext.response() para tocar siempre en un único lugar y no estar buscando por el codigo donde se maneja la resp

judtoro committed:
Future composition is used to deploy verticles on unit tests.
